lib/session.ts

```diff
@@ -228,11 +235,29 @@ export class Session {
   }
 
   private initializeOps(graph: Graph): void {
+    graph.partitionBy(this.supportedOps);
     const nodes = graph.getNodes();
+    const values = graph.getValues();
     this._ops = new Array(nodes.length);
 
     for (let i = 0; i < nodes.length; i++) {
-      this._ops[i] = this.sessionHandler.resolve(nodes[i], this._model.opsets);
+      if (nodes[i].opType === 'NNSubgraph') {
+        const subgraph = nodes[i] as NNSubgraphNode;
+        const initializers = new Map<number, Tensor>();
+
+        for (const subgraphNode of subgraph.nodes) {
+          for (const input of subgraphNode.inputs) {
+            const tensor = values[input].tensor;
+            if (tensor !== undefined) {
+              initializers.set(input, tensor);
+            }
+          }
+        }
+
+        this._ops[i] = new NNSubgraph(subgraph, initializers, this.enablePseudoReorder, this.prefer, this.profiler);
```

Comment:
And we can encapsulate above code into `sessionHandler.resolve` of WebNN backend.